distributedsocialnetwork/post/retrieval.py

```diff
@@ -177,6 +177,61 @@ def get_detailed_post(post_id):
     # response = requests.get(
     #             url, headers={'content-type': 'application/json', 'Accept': 'application/json'})
 
+def get_comments(pk):
+    #return Comment.objects.filter(post_id=post_id)
+    comments = Comment.objects.filter(post_id=pk)
+    comm_ids = []
+    for comment in comments:
+        comm_ids.append(comment["id"])
+    nodes = list(Node.objects.all())
+    for node in nodes:
+        if node.node_auth_username != "":
+            # We can authenticate
+            url = node.api_url + 'posts/' + pk
+            print("sending to ", url)
+            # Include a header containing the Author's id, other nodes may not conform to this.
+            response = requests.get(
+                url, auth=(node.node_auth_username, node.node_auth_password), headers={'content-type': 'application/json', 'Accept': 'application/json'})
+            if response.status_code == 200:
+                #If they have it, check the comments for any uncached comments
+                posts_json = response.json()
+                for post in posts_json["posts"]:
+                    post = sanitize_post(post)
+                    for comment in post.comments:
+                        # Add comment to list of comments to return if it is not already there
+                        if comment["id"] not in comm_ids:
+                            comments.append(comment)
+                            author = sanitize_author(comment["author"])
```

Comment:
This assumes that the author is in our database. It might not be. We should be calling get_detailed_author here.